### PR TITLE
Implement `spy_return_list`.

### PR DESCRIPTION
**Pull request by frank-lenormand**
Related to #378

### CODE REVIEW

File: tests/test_pytest_mock.py [hunk 1]
```diff
@@ -279,8 +279,12 @@ def bar(self, arg):
     assert other.bar(arg=10) == 20
     foo.bar.assert_called_once_with(arg=10)  # type:ignore[attr-defined]
     assert foo.bar.spy_return == 20  # type:ignore[attr-defined]
+    assert foo.bar.spy_return_list == [20]  # type:ignore[attr-defined]
     spy.assert_called_once_with(arg=10)
     assert spy.spy_return == 20
+    assert foo.bar(arg=11) == 22
+    assert foo.bar(arg=12) == 24
+    assert spy.spy_return_list == [20, 22, 24]
```
Comment (nicoddemus) [inline tests/test_pytest_mock.py:287]: For completeness:

```suggestion
    assert spy.spy_return == 24
```

File: docs/usage.rst [hunk 2]
```diff
@@ -79,7 +79,8 @@ are available (like ``assert_called_once_with`` or ``call_count`` in the example
 
 In addition, spy objects contain two extra attributes:
 
-* ``spy_return``: contains the returned value of the spied function.
+* ``spy_return``: contains the last returned value of the spied function.
+* ``spy_return_list``: contains a list of all returned value of the spied function.
```
Comment (nicoddemus) [inline docs/usage.rst:83]:
```suggestion
* ``spy_return_list``: contains a list of all returned value of the spied function (new in ``3.13``).
```

File: src/pytest_mock/plugin.py [hunk 3]
```diff
@@ -43,16 +46,55 @@ class PytestMockWarning(UserWarning):
...
 
 
+@dataclass
+class MockCacheItem:
+    mock: MockType
+    patch: Optional[Any] = None
+
+
+@dataclass
+class MockCache:
```
Comment (nicoddemus) [inline src/pytest_mock/plugin.py:56]: Loved this refactoring. :+1:

**Review by nicoddemus — APPROVED** [review]
Thanks a lot @frank-lenormand!

I will merge and make a new release later!

File: docs/usage.rst [hunk 4]
```diff
@@ -79,7 +79,8 @@ are available (like ``assert_called_once_with`` or ``call_count`` in the example
 
 In addition, spy objects contain two extra attributes:
 
-* ``spy_return``: contains the returned value of the spied function.
+* ``spy_return``: contains the last returned value of the spied function.
+* ``spy_return_list``: contains a list of all returned value of the spied function (new in ``3.13``).
```
Comment (nicoddemus) [inline docs/usage.rst:83]:
```suggestion
* ``spy_return_list``: contains a list of all returned values of the spied function (new in ``3.13``).
```

**Comment by frank-lenormand** [comment]
Thank you!

**Pull request merged**